recommender/engines/base/base_inference_component.py

```diff
@@ -17,20 +17,42 @@
 
 class BaseInferenceComponent(ABC):
     """
-    Base Recommender class with basic functionality
+    Base Recommender class with basic functionality.
+    Used in all available recommender engines.
     """
 
+    engine_name = None
+
     def __init__(self, K: int) -> None:
         self.K = K
         if self.K not in PANEL_ID_TO_K.values():
             raise InvalidRecommendationPanelIDError()
+
+    def _get_recommendation_context(
+        self, context: Dict[str, Any]
+    ) -> [Tuple[int], SearchData]:
+        elastic_services = tuple(context.get("elastic_services") or ())
+        search_data = context.get("search_data")
+
+        if len(elastic_services) < self.K:
+            raise InsufficientRecommendationSpaceError()
+
+        return elastic_services, search_data
+
+
+class MLEngineInferenceComponent(BaseInferenceComponent):
+    """
+    Recommender class for machine learning engines that serves recommendations to logged-in users.
+    Used in NCF and RL engines.
+    """
+
+    def __init__(self, K: int) -> None:
+        super().__init__(K)
         self._load_models()
 
     def __call__(self, context: Dict[str, Any]) -> List[int]:
```

Comment:
can you add abstract `__call__` with this interface to the BaseInferenceComponent? (with empty implementation ofc)